### PR TITLE
Basic `/help` functionality

### LINKED ISSUE #8: Slash Command - basic Help
PR at #23

### PR DESCRIPTION
**Pull request by AverageHelper**
At its very basic, `/help` lists the commands available to the user. This is at least a start.

Screen Shot 2022-09-28 at 23 23 19

This command will automatically include new commands as they're added in `src/commands/index.ts`.

Closes #8

### CODE REVIEW

File: src/commands/help.ts
Comment (gmacgre) [inline src/commands/help.ts:28]: Is this TODO being worked on at the moment?

File: src/commands/help.ts
Comment (gmacgre) [inline src/commands/help.ts:35]: Looks like this section is not covered in tests. We should probably make sure that it works in DMs as well before merging.

**Review by gmacgre — CHANGES_REQUESTED** [review]
Other than those two small fixes, LGTM

File: src/commands/help.ts
Comment (gmacgre) [reply src/commands/help.ts:28]: Or is the building of perms checking something that we should add to the docket and circle back to?

Comment (AverageHelper) [reply src/commands/help.ts:28]: Might be complicated. I poked at the problem a bit, and I realized I need to learn how this new guild-centric permission system works. I'm used to the bot maintaining a database of its own command permissions, but slash commands don't do that anymore, so I need to re-learn how the APIs relate to the UI.

If someone can pick that up before I get around to it, be my guest. Meantime, since Discord enforces command permissions anyway, I'm fine pushing this one down the road.

Comment (gmacgre) [reply src/commands/help.ts:28]: Alright. I think we can revisit this when we get a little more experience with the API then. I'll add is as a ticket for future building.

File: src/commands/help.ts
Comment (AverageHelper) [reply src/commands/help.ts:35]: Good point. Adding a test now

**Review by gmacgre — APPROVED** [review]
LGTM. I think we have everything we need now for this command.

**Comment by gmacgre** [comment]
Note- We may want to wait for one more person to approve this before merging.

File: src/commands/__snapshots__/help.test.ts.snap
Comment (JstnMcBrd) [inline src/commands/__snapshots__/help.test.ts.snap:1]: What's this extra file all about? And the weird `__snapshots__` subdirectory? Clutter?

Comment (AverageHelper) [reply src/commands/__snapshots__/help.test.ts.snap:1]: That's a smapshot file, generated by Jest the first time a snapshot is taken. Future tests will compare outputs against this file, and yell if things change unexpectedly, indicating a failed test.

**Review by JstnMcBrd — APPROVED** [review]

**Comment by JstnMcBrd** [comment]
Oh I forgot to mention, probably should keep the changelog up to date.

**Comment by AverageHelper** [comment]
Fair point

**Pull request merged**
